client/src/App.js

```diff
@@ -1,11 +1,135 @@
+import React, { useState, useEffect } from "react";
 import "./App.css";
+import Videocard from "./Videocard";
+import AddingItem from "./AddingItem";
+import { ErrorBoundary } from "react-error-boundary";
 
 function App() {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [updated, setUpdated] = useState(true);
+  const [sortOrder, setSortOrder] = useState("desc");
+
+  useEffect(() => {
+    fetch("https://videos-1rb3.onrender.com/")
+      .then((response) => response.json())
+      .then((data) => {
+        setVideos(data);
+        setLoading(false);
+    })
+      .catch((error) => console.log(error));
+  }, [updated]);
+
+  const addVideo = (newVideo) => {
+    fetch("https://videos-1rb3.onrender.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newVideo),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("Response:", data);
+        setUpdated(!updated);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  };
+
+  const upVote = (video) => {
+    const updatedVideos = videos.map((v) =>
+      v === video ? { ...v, rating: v.rating + 1 } : v
+    );
+    setVideos(updatedVideos);
+    
+    fetch(`https://videos-1rb3.onrender.com/update_rating`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ id: video.id, rating: video.rating + 1 }),
+    })
+      .then((response) => response.json())
+      .then((data) => console.log(data))
+      .catch((error) => console.log("Error: ", error));
+  }
+
+  const downVote = (video) => {
```

Comment:
This function looks very similar to the `upVote` function - can you think of a way to make them share code rather than duplicate it?